tests/rbd_mirror/test_rbd_mirror_shrink_image_primary.py

```diff
@@ -0,0 +1,131 @@
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror.rbd_mirror_utils import rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_image_shrink_primary(rbd_mirror, pool_type, **kw):
+    """
+    Test to verify image size in secondary when shrink in primary
+
+    Args:
+        rbd_mirror: rbd mirror object
+        pool_type: ec pool or rep pool
+        **kw: test data
+        Example::
+            ec_pool_config:
+                mode: image
+                mirrormode: snapshot
+                size: 2G
+                io_total: 10
+
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+    """
+    try:
+        log.info(kw)
+        mirror1 = rbd_mirror.get("mirror1")
+        mirror2 = rbd_mirror.get("mirror2")
+        config = kw.get("config")
+        pool = config[pool_type]["pool"]
+        image = config[pool_type]["image"]
+        imagespec = pool + "/" + image
+        resize = "1G"
+        mirror1.resize_image(imagespec=imagespec, size=resize)
+
+        if config[pool_type]["mode"] == "image":
+            mirror1.create_mirror_snapshot(imagespec)
+            mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+        else:
+            mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+        out1 = rbd1.image_info(pool, image)
+        out2 = rbd2.image_info(pool, image)
+        if out1["size"] == out2["size"]:
+            log.info(f"New image size {resize} is updated in secondary")
+            return 0
+        log.error(f"New image size {resize} is not updated in secondary")
+        return 1
+
+    except Exception as e:
+        log.exception(e)
+        return 1
```

Comment:
I think we could use run method doc if we don't have anything special (explicit) to perform here in exception block
BAsed on no explicit operation in exception block, can we make use of run method's exception block ?